### PR TITLE
Modify S2230: extends with all Spring proxy

### PR DESCRIPTION
**Pull request by erwan-serandour-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S2230/java/rule.adoc [hunk 1]
```diff
@@ -1,57 +1,62 @@
 == Why is this an issue?
 
-Marking a non-`public` method `@Transactional` is both useless and misleading because Spring does not recognize non-`public` methods, and so makes no provision for their proper invocation.
-Nor does Spring make provision for the methods invoked by the method it called.
+A method annotated with Spring's `@Async` or `@Transactional` annotations will not work as expected
+if invoked directly from within its class.
```
Comment (dorian-burihabwa-sonarsource) [inline rules/S2230/java/rule.adoc:4]: To track changes in the documentation, we should start and end sentences on the same line
```suggestion
A method annotated with Spring's `@Async` or `@Transactional` will not work as expected if invoked directly from within its class.
```

File: rules/S2230/java/metadata.json [hunk 2]
```diff
@@ -1,32 +1,26 @@
 {
-  "title": "Non-public methods should not be \"@Transactional\"",
+  "title": "Methods with Spring proxy should be public",
```
Comment (dorian-burihabwa-sonarsource) [inline rules/S2230/java/metadata.json:2]:
```suggestion
  "title": "Methods with Spring proxying annotations should be public",
```

File: rules/S2230/java/rule.adoc [hunk 3]
```diff
@@ -1,57 +1,62 @@
...
 
+This is because Spring generates a proxy class with wrapper code to manage the method's asynchronicity (`@Async`)
+or to handle the transaction (`@Transactional`).
+However, when called using `this`, the proxy instance is bypassed, and the method is invoked directly
+without the required wrapper code.
```
Comment (dorian-burihabwa-sonarsource) [inline rules/S2230/java/rule.adoc:9]: This should be covered by another rule, let's remove it for now

File: rules/S2230/java/metadata.json [hunk 4]
```diff
@@ -1,32 +1,26 @@
...
-
-    ],
-    "legacyKeys": [
-
-    ]
+  "code": {
+    "impacts": {
+      "MAINTAINABILITY": "HIGH",
+      "RELIABILITY": "MEDIUM",
+      "SECURITY": "LOW"
```
[the remark this answers is not in the document]
Comment (erwan-serandour-sonarsource) [reply rules/S2230/java/metadata.json:16]: :+1:

File: rules/S2230/java/metadata.json [hunk 5]
```diff
@@ -1,32 +1,26 @@
...
+    },
+    "attribute": "CONVENTIONAL"
   },
   "defaultSeverity": "Major",
   "ruleSpecification": "RSPEC-2230",
   "sqKey": "S2230",
-  "scope": "Main",
+  "scope": "All",
```
[the remark this answers is not in the document]
Comment (erwan-serandour-sonarsource) [reply rules/S2230/java/metadata.json:23]: :+1:

File: rules/S2230/java/metadata.json (hunk 2 above)
Comment (erwan-serandour-sonarsource) [reply rules/S2230/java/metadata.json:2]: :+1:

File: rules/S2230/java/rule.adoc [hunk 6]
```diff
@@ -1,57 +1,53 @@
 == Why is this an issue?
 
-Marking a non-`public` method `@Transactional` is both useless and misleading because Spring does not recognize non-`public` methods, and so makes no provision for their proper invocation.
-Nor does Spring make provision for the methods invoked by the method it called.
+Marking a non-public method `@Async` or `@Transactional` is both useless and misleading because Spring does not recognize non-public methods, and so makes no provision for their proper invocation. Nor does Spring make provision for the methods invoked by the method it called.
```
Comment (dorian-burihabwa-sonarsource) [inline rules/S2230/java/rule.adoc:3]: Let's break the line before the second sentence

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]
Thanks for the chanes

**Pull request merged**